### PR TITLE
Update replace-image-registry.yaml to no use defaultNamespace for non…

### PR DESCRIPTION
**Pull request by dunefro**
… docker.io registry

### CODE REVIEW

File: charts/tfy-jspolicy-configs/templates/replace-image-registry.yaml [hunk 1]
```diff
@@ -95,13 +102,14 @@ spec:
...
       let registryReplacement = registryReplacementMap[registry];
       if (!registryReplacement) {
+        if (!registryReplacementMap['*']) {
+          return imageUri;
+        }
         registryReplacement = registryReplacementMap['*'];
-      }
-      if (imageUri.startsWith(registryReplacement)) {
-        return imageUri
       } else {
```
Comment (shubhamrai1993) [inline charts/tfy-jspolicy-configs/templates/replace-image-registry.yaml:109]: What is this else condition for?

Comment (dunefro) [reply charts/tfy-jspolicy-configs/templates/replace-image-registry.yaml:109]: Else condition is for the case where registry is in the map. First if condition is when registry to be replaced is not in map.

Comment (shubhamrai1993) [reply charts/tfy-jspolicy-configs/templates/replace-image-registry.yaml:109]: But its already assigned the same value in line #96. So this condition shouldn't be needed

File: charts/tfy-jspolicy-configs/templates/replace-image-registry.yaml [hunk 2]
```diff
@@ -95,13 +102,14 @@ spec:
       let newImageUri = imageUri;
       let registryReplacement = registryReplacementMap[registry];
       if (!registryReplacement) {
+        if (!registryReplacementMap['*']) {
+          return imageUri;
+        }
         registryReplacement = registryReplacementMap['*'];
-      }
-      if (imageUri.startsWith(registryReplacement)) {
```
Comment (shubhamrai1993) [inline charts/tfy-jspolicy-configs/templates/replace-image-registry.yaml:100]: Why remove this check? This was needed to prevent recursive replacement

**Review by shubhamrai1993 — APPROVED** [review]

**Pull request merged**